addons/website/static/src/js/tours/tour_generator.js

```diff
@@ -0,0 +1,114 @@
+         * @param {*} snippet contains an id, a name and could have options
+         */
+        addSnippet(snippet){
+            this.snippets.push(snippet);
+        }
+
+        /**
+         * Render the tour generation using this.snippets
+         */
+        render(){
+            this.snippets.forEach(snippet => {
+                // each snippet need to be drag & drop
+                this.dragAndDropGeneration(snippet);
+                
+                if (snippet.options){
+                    // For each options, use the appropriate function to render its step
+                    Object.keys(snippet.options).forEach(key => {
+                        this.stepByOptions[key](snippet, this);
+                    });
+                    // Then go back to blocks.
+                    // OSU TODO : Useless for the last snippet if he has options
+                    this.goBackToBlocks();
+                }
+            });
+            // each theme tour will end by saving the website
+            this.clickSave();
+            return this.allSteps;
+        }
+
+        dragAndDropGeneration(snippet){
+            this.allSteps.push({
+                trigger: `#snippet_structure .oe_snippet[name="${snippet.name}"] .oe_snippet_thumbnail`,
+                extra_trigger: "body.editor_enable.editor_has_snippets",
+                content: _t(`Drag the <i>${snippet.name}</i> block and drop it in your page.`),
+                position: "bottom",
+                run: "drag_and_drop #wrap",
+            });
+        }
+
+        textGeneration(snippet, source){
+            // Because it comes from an object, this doesn't target the class.
+            source ? source = source : source = this;
+            source.allSteps.push({
+                trigger: `#wrapwrap .${snippet.id} ${snippet.options['text'].element}`,
+                content: _t("<b>Click on a text</b> to start editing it. <i>It's that easy to edit your content!</i>"),
+                position: "bottom",
+                run: "text",
+            });
+        }
+
+        backgroundGeneration(snippet, source){
+            source ? source = source : source = this;
+            source.allSteps.push({
+                trigger: ".o_we_customize_panel .o_we_edit_image",
+                content: _t("Customize any block through this menu. Try to change the background color of this block."),
+                position: "bottom",
+            });
+        }
+
+        goBackToBlocks(){
+            this.allSteps.push({
+                trigger: '.o_we_add_snippet_btn',
+                content: _t("Go back to the blocks menu."),
+                position: 'bottom',
+            });
+        }
+
+        clickSave(){
+            this.allSteps.push({
+                trigger: "button[data-action=save]",
+                content: _t("Click the <b>Save</b> button."),
+                position: "bottom",
+            });
+        }
+    }
+
+    return TourThemeGenerator;
```

Comment:
Ok, we don't need a class to handle this :wink: We just need to expose all the functions you created. So here instead of returning a generator class, you should return an object with a series of utility methods to create steps we need to create those tours.

So when we create a tour, instead of
```javascript
tour.register(name, [{
   // ...
}, {
   // ...
}, {
   // ...
}]);
```

We would have to do:
```javascript
const wTourUtils = require('website.tour.utils');

tour.register(name, [{
    // ...
},
...wTourUtils.dropSnippet('s_title'),
{
    // ...
}]);
```

And each function returned by `website.tour.utils` would return a series of tour steps according to given parameters. These functions can then be reused by any tour (not only theme tours) which deals with the editor.

Reply:
I just pushed the modifications, feel free to check it and review it :)